File: frontend/autocare-frontend/src/services/invoiceService.js
import api from "./api";

const invoiceService = {

    // Get all invoices
    getAllInvoices() {
        return api.get("/invoices");
    },

    // Get invoice by ID
    getInvoiceById(id) {
        return api.get(`/invoices/${id}`);
    },

    // Get invoice by invoice number
    getInvoiceByNumber(invoiceNumber) {
        return api.get(`/invoices/number/${invoiceNumber}`);
    },

    // Create invoice
    createInvoice(invoice) {
        return api.post("/invoices", invoice);
    },

    // Delete invoice
    deleteInvoice(id) {
        return api.delete(`/invoices/${id}`);
    }

};

export default invoiceService;

File: autocare-backend/src/main/java/com/pobitra/autocare/service/InvoiceService.java
package com.pobitra.autocare.service;

import com.pobitra.autocare.dto.InvoiceRequestDTO;
import com.pobitra.autocare.dto.InvoiceResponseDTO;

import java.util.List;

public interface InvoiceService {

    InvoiceResponseDTO createInvoice(InvoiceRequestDTO dto);

    List<InvoiceResponseDTO> getAllInvoices();

    InvoiceResponseDTO getInvoiceById(Long id);

    InvoiceResponseDTO getInvoiceByNumber(String invoiceNumber);

    InvoiceResponseDTO updateInvoice(Long id, InvoiceRequestDTO dto);

    void deleteInvoice(Long id);
}